org.jacoco.agent.rt/src/org/jacoco/agent/rt/internal/Agent.java

```diff
@@ -110,21 +114,19 @@ public RuntimeData getData() {
 	/**
 	 * Initializes this agent.
 	 * 
+	 * @throws Exception
+	 *             in case something cannot be initialized
 	 */
-	public void startup() {
-		try {
-			String sessionId = options.getSessionId();
-			if (sessionId == null) {
-				sessionId = createSessionId();
-			}
-			data.setSessionId(sessionId);
-			output = createAgentOutput();
-			output.startup(options, data);
-			if (options.getJmx()) {
-				jmxRegistration = new JmxRegistration(this);
-			}
-		} catch (final Exception e) {
-			logger.logExeption(e);
```

Comment:
@marchof maybe we should keep this output in addition to throwing of exception for the case when caller swallows exception, e.g. offline instrumented classes of app inside of app-server? WDYT?

Reply:
@Godin Perfectly valid point. I restored logging.